streams/src/main/java/org/apache/kafka/streams/kstream/internals/StatefulProcessorNode.java

```diff
@@ -28,40 +26,38 @@
 class StatefulProcessorNode<K, V> extends StatelessProcessorNode<K, V> {
 
     private final String[] storeNames;
-    private final org.apache.kafka.streams.processor.StateStoreSupplier<KeyValueStore> storeSupplier;
     private final StoreBuilder<KeyValueStore<K, V>> storeBuilder;
+    private final String maybeRepartitionedSourceName;
 
 
-    StatefulProcessorNode(final String parentNodeName,
-                          final String processorNodeName,
-                          final ProcessorSupplier<K, V> processorSupplier,
+    StatefulProcessorNode(final String nodeName,
+                          final ProcessorParameters processorParameters,
                           final String[] storeNames,
-                          final org.apache.kafka.streams.processor.StateStoreSupplier<KeyValueStore> storeSupplier,
+                          final String maybeRepartitionedSourceName,
                           final StoreBuilder<KeyValueStore<K, V>> storeBuilder,
                           final boolean repartitionRequired) {
-        super(parentNodeName,
-              processorNodeName,
-              processorSupplier,
+        super(nodeName,
+              processorParameters,
               repartitionRequired);
 
         this.storeNames = storeNames;
-        this.storeSupplier = storeSupplier;
         this.storeBuilder = storeBuilder;
+        this.maybeRepartitionedSourceName = maybeRepartitionedSourceName;
     }
 
 
     String[] storeNames() {
         return Arrays.copyOf(storeNames, storeNames.length);
     }
 
-    org.apache.kafka.streams.processor.StateStoreSupplier<KeyValueStore> storeSupplier() {
-        return storeSupplier;
-    }
-
     StoreBuilder<KeyValueStore<K, V>> storeBuilder() {
         return storeBuilder;
     }
 
+    String maybeRepartitionedSourceName() {
```

Comment:
This function not used.

Reply:
I'd like to leave in for now and during the 3rd phase I'll remove it if it's not needed.

Reply:
I agree with Guozhang.